src/main/java/core/basesyntax/strategy/impl/SupplyOperation.java

```diff
@@ -0,0 +1,12 @@
+package core.basesyntax.strategy.impl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.strategy.OperationHandler;
+
+public class SupplyOperation implements OperationHandler {
+    @Override
+    public void handle(FruitTransaction transaction) {
+        Storage.addFruit(transaction.getFruit(), transaction.getQuantity());

```

Comment:
Ensure that the 'addFruit' method in the 'Storage' class properly encapsulates the logic for adding fruit, such as checking for null values and that the quantity is not negative. Encapsulation is important for maintaining the integrity of the data within the 'Storage'.